src/lib/comms/sol-oic-server.c

```diff
@@ -754,6 +681,21 @@ sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
     return NULL;
 }
 
+SOL_API struct sol_oic_server_resource *
+sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
+    const void *handler_data, enum sol_oic_resource_flag flags)
```

Comment:
Alternatively, why not add the path to the ```struct sol_oic_resource_type``` and, if empty, just generate one. No need for two APIs then. Wouldn't even break ABI if the path is at the end of the structure.